ios/RNBackgroundDownloader.m

```diff
@@ -79,17 +79,21 @@ - (void)lazyInitSession {
 }
 
 - (void)removeTaskFromMap: (NSURLSessionTask *)task {
-    NSNumber *taskId = @(task.taskIdentifier);
-    RNBGDTaskConfig *taskConfig = taskToConfigMap[taskId];
-    [taskToConfigMap removeObjectForKey:taskId];
-    [[NSUserDefaults standardUserDefaults] setObject:[self serialize: taskToConfigMap] forKey:ID_TO_CONFIG_MAP_KEY];
-    if (taskConfig) {
-        [idToTaskMap removeObjectForKey:taskConfig.id];
-        [idToPercentMap removeObjectForKey:taskConfig.id];
-    }
-    if (taskToConfigMap.count == 0) {
-        [urlSession invalidateAndCancel];
-        urlSession = nil;
+    @synchronized (sharedLock) {
```

Comment:
@mark-plukkido Just curious, was it your intention to reverse this commit: https://github.com/EkoLabs/react-native-background-downloader/pull/61/commits/44eaa8c87130bd5c68a37e9c107773e42a443d07?

Reply:
No, absolutely not. I made a mistake when I merged from upstream and then in 5d86a50 I changed it back to your code. There should be only one change in RNBackgroundDownloader.java. 

Do you want me to create a new PR without these? In that case, just reject this PR and I recreate it without the extra confusion.

Thanks!

Reply:
@ptelad Could you reject this PR so @mark-plukkido Can recreate it? I don't think I have the ability to reject as I am not a contributor/owner in this repo. I was just looking at the latest changes and PRs and I noticed something was off with this one.

Reply:
@mark-plukkido Oh I see what happened. I was just looking through these new PRs and got a little confused. Its probably up to you and @ptelad whether or not you'd like to remake the PR.